pkg/customresourcestate/registry_factory.go

```diff
@@ -674,14 +780,85 @@ func famGen(f compiledFamily) generator.FamilyGenerator {
 	}
 }
 
+func findWildcard(path valuePath, i *int) bool {
+	for ; *i < len(path); *i++ {
+		if path[*i].part == "*" {
+			return true
+		}
+	}
+	return false
+}
+
+func resolveWildcard(path valuePath, object map[string]interface{}) []valuePath {
+	if path == nil {
+		return nil
+	}
+	checkpoint := object
+	var expandedPaths []valuePath
+	var list []interface{}
+	var l int
+	for i, j := 0, 0; findWildcard(path, &i); /* i is at "*" now */ {
+		for ; j < i; j++ {
+			maybeCheckpoint, ok := checkpoint[path[j].part]
+			if !ok {
+				// path[j] is not in the object, so we can't expand the wildcard
+				return []valuePath{path}
+			}
+			// store (persist) last checkpoint
+			if c, ok := maybeCheckpoint.([]interface{}); ok {
+				list = c
+				break
+			}
+			checkpoint = maybeCheckpoint.(map[string]interface{})
+		}
+		if j > i {
+			break
+		}
+		// i is at "*", j is at the last part before "*", checkpoint is at the value of the last part before "*"
+		l = len(list) // number of elements in the list
+		pathCopyStart := make(valuePath, i)
+		copy(pathCopyStart, path[:i])
+		pathCopyWildcard := make(valuePath, len(path)-i-1)
+		copy(pathCopyWildcard, path[i+1:])
+		for k := 0; k < l; k++ {
+			pathCopyStart = append(pathCopyStart, pathOp{part: strconv.Itoa(k)})
+			pathCopyStart = append(pathCopyStart, pathCopyWildcard...)
+			expandedPaths = append(expandedPaths, pathCopyStart)
+		}
```

Comment:
since we know the size of the path from the get go, it would be less expensive in terms of allocations to just allocate one slice and then fill it correctly